src/Pulsar.Client/Api/Producer.fs

```diff
@@ -69,9 +70,19 @@ type Producer private (producerConfig: ProducerConfiguration, lookup: BinaryLook
                         channel.Reply()
                         return! loop state
                     | NotConnected ->
-                        //TODO put message on schedule
-                        Log.Logger.LogWarning("NotConnected, skipping send")
+                        pendingMessages.Enqueue msg
```

Comment:
Channel is not saved anywhere so producer's user won't ever receive response once pending messages will be reprocessed.